examples/energy-management-app/energy-management-common/include/DeviceEnergyManagementDelegateImpl.h

```diff
@@ -18,69 +18,276 @@
 
 #pragma once
 
-#include "app/clusters/device-energy-management-server/device-energy-management-server.h"
-
+#include <app/clusters/device-energy-management-server/device-energy-management-server.h>
 #include <app/util/config.h>
-#include <cstring>
 
 using chip::Protocols::InteractionModel::Status;
 namespace chip {
 namespace app {
 namespace Clusters {
 namespace DeviceEnergyManagement {
 
+class DEMManufacturerDelegate;
+
 /**
  * The application delegate.
  */
 class DeviceEnergyManagementDelegate : public DeviceEnergyManagement::Delegate
 {
 public:
+    DeviceEnergyManagementDelegate();
+
+    void SetDeviceEnergyManagementInstance(DeviceEnergyManagement::Instance & instance);
+
+    void SetDEMManufacturerDelegate(DEMManufacturerDelegate & deviceEnergyManagementManufacturerDelegate);
+
+    /**
+     *
+     * Implement the DeviceEnergyManagement::Delegate interface
+     *
+     */
+
+    /**
+     * @brief Implements a handler to begin to adjust client power
+     *        consumption/generation to the level requested.
+     *
+     * @param power Milli-Watts the ESA SHALL use during the adjustment period.
+     * @param duration The duration that the ESA SHALL maintain the requested power for.
+     * @return  Success if the adjustment is accepted; otherwise the command SHALL be rejected with appropriate error.
+     */
     virtual Status PowerAdjustRequest(const int64_t power, const uint32_t duration, AdjustmentCauseEnum cause) override;
+
+    /**
+     * @brief Make the ESA end the active power adjustment session & return to normal (or idle) power levels.
+     *        The ESA SHALL also generate an PowerAdjustEnd Event and the ESAState SHALL be restored to Online.
+     *
+     * @return It should report SUCCESS if successful and FAILURE otherwise.
+     */
     virtual Status CancelPowerAdjustRequest() override;
+
+    /**
+     * @brief The ESA SHALL update its Forecast attribute with the RequestedStartTime including a new ForecastID.
+     *
+     *   If the ESA supports ForecastAdjustment, and the ESAState is not UserOptOut and the RequestedStartTime is after
+     *   the EarliestStartTime and the resulting EndTime is before the LatestEndTime, then ESA SHALL accept the request
+     *   to modify the Start Time.
+     *   A client can estimate the entire Forecast sequence duration by computing the EndTime - StartTime fields from the
+     *   Forecast attribute, and therefore avoid scheduling the start time too late.
+     *
+     * @param requestedStartTime The requested start time in UTC that the client would like the appliance to shift its power
```

Comment:
requestedStartTimeUTC

Reply:
Should the UTC be all caps or Utc? I have added Mw to power above -> powerMw - I am looking around other clusters for example styles to follow and I found timeoutMs (milli-seconds) so am following that convention.

OOC are there any code standard guidelines that I can follow pls?

Reply:
The reason why these variables didn’t have mW etc is that all of these types are standard types (always mW/mA/mV) and epoch_s.

The variables are also intended to match the attribute names which do not include _mW etc.

Where we deviate from mW/mA/mV (or for local internal variables etc) then absolutely we should highlight that the values type. But the original code for attributes explicitly didn’t rename them like this.

Reply:
We have had numerous bugs for missing unit of measurement. if you want to use spec names, then please find a way to define a unit-safe data type (i.e. not a integer type or an aliased integer type). Otherwise i recomment just placing units in names for C++ code.